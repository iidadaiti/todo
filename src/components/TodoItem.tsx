import React, { ComponentProps, FC } from 'react'
import { TodoData } from './Todo'

export interface TodoItemProps extends Omit<ComponentProps<'div'>, 'children'> {
  data: TodoData
  onRemoveTodo?: (todoData: TodoData) => void
  onChangeTodo?: (before: TodoData, after: TodoData) => void
}

export const TodoItem: FC<TodoItemProps> = ({
  data,
  onRemoveTodo,
  onChangeTodo,
  ...props
}) => {
  return (
    <label>
      <div {...props}>
        <input
          type="checkbox"
          checked={data.isDone}
          onChange={() =>
            onChangeTodo &&
            onChangeTodo(data, { ...data, isDone: !data.isDone })
          }
        />
        <span>{data.text}</span>
        <button
          onClick={() => {
            onRemoveTodo && onRemoveTodo(data)
          }}
          disabled={!onRemoveTodo}
        >
          remove
        </button>
      </div>
    </label>
  )
}
