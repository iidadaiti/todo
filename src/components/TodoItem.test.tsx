import { render } from '@testing-library/react'
import React from 'react'
import { TodoItem } from './TodoItem'

test('TodoItem snapshot', () => {
  const data = {
    key: 'test',
    text: 'test',
    isDone: false,
    createAt: new Date()
  }

  const result = render(<TodoItem data={data} />)
  expect(result.asFragment()).toMatchSnapshot()
})
