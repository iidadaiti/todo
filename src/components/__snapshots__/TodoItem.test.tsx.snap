// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem snapshot 1`] = `
<DocumentFragment>
  <label>
    <div>
      <input
        type="checkbox"
      />
      <span>
        test
      </span>
      <button
        class="spectrum-Button spectrum-Button--cta spectrum-Button--sizeM"
        disabled=""
      >
        <span
          class="spectrum-Button-label"
        >
          Remove
        </span>
      </button>
    </div>
  </label>
</DocumentFragment>
`;
